scripts/generate-releases.sh

```diff
@@ -16,22 +17,46 @@ fi
 ENVIRONMENT="$1"
 RELEASE_NAME_SUFFIX="$2"
 OPERATOR_INDEX_COMMIT="${3:-$(git rev-parse HEAD)}"
+OPERATOR_INDEX_BRANCH="${4:-$(git rev-parse --abbrev-ref HEAD)}"
+
+# Fetch the list of snapshots for OPERATOR_INDEX_COMMIT and OPERATOR_INDEX_BRANCH. 
+# Make sure that only one snapshot per application is returned, and that the snapshots are sorted by creation timestamp.
+snapshot_list="$(kubectl get snapshot -l pac.test.appstudio.openshift.io/sha="${OPERATOR_INDEX_COMMIT}" -o json | jq -r '
+  .items
+  | map(select(.metadata.annotations["pac.test.appstudio.openshift.io/branch"]=="'"${OPERATOR_INDEX_BRANCH}"'"))
+  | sort_by(.spec.application)
+  | group_by(.spec.application)
+  | map(sort_by(.metadata.creationTimestamp) | last)
+  | .[]
+  | "\(.metadata.name)|\(.spec.application)"
+')"
 
 validate_input() {
-    if [ "$(kubectl get snapshot -l pac.test.appstudio.openshift.io/sha="${OPERATOR_INDEX_COMMIT}" --no-headers | wc -l)" -eq 0 ]; then
-        echo "ERROR: Could not find any Snapshots for the commit '${OPERATOR_INDEX_COMMIT}'. This must be a 40 character-long commit SHA. Default: currently checked out commit."
+    supported_ocp_number=$(find ".tekton" -maxdepth 1 -type f -name "operator-index-ocp-*-build.yaml" | wc -l )
+    snapshot_number=$(echo "$snapshot_list" | wc -l )
```

Comment:
[nits]
1. Should these be called `pipelines_count` and `snapshots_count`?
2. Please don't forget to quote.

```suggestion
    pipelines_count="$(find ".tekton" -maxdepth 1 -type f -name "operator-index-ocp-*-build.yaml" | wc -l )"
    snapshots_count="$(echo "$snapshot_list" | wc -l )"
```